test/e2e/pvc_label_selector.go

```diff
@@ -0,0 +1,159 @@
+   2. Create pv with lable volume-type:ssd, volume path set to vmdk created in previous step, and PersistentVolumeReclaimPolicy is set to Delete.
+   3. Create PVC (pvc_vvol) with label selector to match with volume-type:vvol
+   4. Create PVC (pvc_ssd) with label selector to match with volume-type:ssd
+   5. Wait and verify pvc_ssd is bound with PV.
+   6. Verify Status of pvc_vvol is still pending.
+   7. Delete pvc_ssd.
+   8. verify associated pv is also deleted.
+   9. delete pvc_vvol
+
+*/
+var _ = framework.KubeDescribe("pvclabelselector", func() {
+	f := framework.NewDefaultFramework("pvclabelselector")
+	var (
+		c          clientset.Interface
+		ns         string
+		pv_ssd     *v1.PersistentVolume
+		pvc_ssd    *v1.PersistentVolumeClaim
+		pvc_vvol   *v1.PersistentVolumeClaim
+		volumePath string
+		ssdlabels  map[string]string
+		vvollabels map[string]string
+		err        error
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(c, framework.TestContext.NodeSchedulableTimeout))
+		ssdlabels = make(map[string]string)
+		ssdlabels["volume-type"] = "ssd"
+		vvollabels = make(map[string]string)
+		vvollabels["volume-type"] = "vvol"
+
+	})
+
+	framework.KubeDescribe("Selector-Label Volume Binding:vsphere", func() {
+		AfterEach(func() {
+			By("Running clean up actions")
+			if framework.ProviderIs("vsphere") {
+				testCleanupVSpherePVClabelselector(c, ns, volumePath, pv_ssd, pvc_ssd, pvc_vvol)
+			}
+		})
+		It("should bind volume with claim for given label", func() {
+
+			framework.SkipUnlessProviderIs("vsphere")
+			volumePath, pv_ssd, pvc_ssd, pvc_vvol, err = testSetupVSpherePVClabelselector(c, ns, ssdlabels, vvollabels)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("wait for the pvc_ssd to bind with pv_ssd")
+			waitOnPVandPVC(c, ns, pv_ssd, pvc_ssd)
+
+			By("Verify status of pvc_vvol is pending")
+			err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimPending, c, ns, pvc_vvol.Name, 3*time.Second, 300*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("delete pvc_ssd")
+			deletePersistentVolumeClaim(c, pvc_ssd.Name, ns)
+
+			By("verify pv_ssd is deleted")
+			err = framework.WaitForPersistentVolumeDeleted(c, pv_ssd.Name, 3*time.Second, 300*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+			volumePath = ""
+
+			By("delete pvc_vvol")
+			deletePersistentVolumeClaim(c, pvc_vvol.Name, ns)
+		})
+	})
+})
+
+func testSetupVSpherePVClabelselector(c clientset.Interface, ns string, ssdlabels map[string]string, vvollabels map[string]string) (volumePath string, pv_ssd *v1.PersistentVolume, pvc_ssd *v1.PersistentVolumeClaim, pvc_vvol *v1.PersistentVolumeClaim, err error) {
+	volumePath = ""
+	By("creating vmdk")
+	vsp, err := vsphere.GetVSphere()
+	Expect(err).NotTo(HaveOccurred())
+	volumePath, err = createVSphereVolume(vsp, nil)
+	if err != nil {
+		_ = fmt.Errorf("failed to create vmdk")
```

Comment:
Does this discard the return from `Errorf`? If so, what's the purpose? If not, is `err` set somehow?